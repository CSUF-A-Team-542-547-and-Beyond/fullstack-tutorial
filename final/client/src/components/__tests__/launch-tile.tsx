import React from 'react';

import { render, cleanup } from '../../test-utils';
import { shallow } from 'enzyme';
import LaunchTile from '../launch-tile';

import renderer from 'react-test-renderer';

describe('Launch Tile', () => {
  // automatically unmount and cleanup DOM after the test is finished.
  afterEach(cleanup);

  it('renders without error', () => {
    shallow(
      <LaunchTile
//      <RaunchTile //A test to see the test fail to be sure our test can fail
        launch={{
          __typename: 'Launch',
          isBooked: false,
          id: '1',
          mission: { name: 'the first one', __typename: 'Mission', missionPatch: null },
          rocket: { name: 'harambe', __typename: 'Rocket', id: '1' },
        }}
      />,
    );
  });

  it('launch tile matches the snapshot', () => {
    const tree = renderer.create(
      <LaunchTile
        launch={{
          __typename: 'Launch',
          isBooked: false,
          id: '1',
          mission: { name: 'the first one', __typename: 'Mission', missionPatch: null },
          rocket: { name: 'harambe', __typename: 'Rocket', id: '1' },
        }}
      />, 
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
