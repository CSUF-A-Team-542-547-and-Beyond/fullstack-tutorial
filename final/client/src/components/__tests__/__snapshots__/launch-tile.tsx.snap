// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch Tile launch tile matches the snapshot 1`] = `
<a
  className="css-10q0vj5"
  href="/launch/1"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(iss.jpg)",
    }
  }
>
  <h3>
    the first one
  </h3>
  <h5>
    harambe
  </h5>
</a>
`;
